zabbix_auto_config/utils.py

```diff
@@ -84,3 +85,65 @@ def read_map_file(path: Union[str, Path]) -> Dict[str, List[str]]:
             )
         _map[key] = values_dedup
     return _map
+
+
+def with_prefix(
+    text: str,
+    prefix: str,
+    separator: str = "-",
+) -> str:
+    """Replaces the prefix of `text` with `prefix`. Assumes the separator
+    between the prefix and the text is `separator` (default: "-").
+
+    Parameters
+    ----
+    text: str
+        The text to format.
+    prefix: str
+        The prefix to add to `text`.
+    separator: str
+        The separator between the prefix and the text.
+
+    Returns
+    -------
+    str
+        The formatted string.
+    """
+    if not all(s for s in (text, prefix, separator)):
+        raise ValueError("Text, prefix, and separator cannot be empty")
+
+    _, _, postfix = text.partition(separator)
+
+    # Unable to split text, nothing to do
+    if not postfix:
+        raise ValueError(
+            f"Could not find prefix in {text!r} with separator {separator!r}"
+        )
+
+    if prefix.endswith(separator) or postfix.startswith(separator):
+        sep = ""
+    else:
+        sep = separator
```

Comment:
I don't think there is a need to handle this (it could be simpler/more clear without this), but it doesn't have a big impact if it's done.

Reply:
Sure, we can omit that but I don't think it increases the complexity _that_ much, and ensures all groups contain the `separator`. In any case, I have replaced it with a warning, so that we are at least able to monitor if it happens.

I think the convention for group name prefixes should be formalized in some way, so that missing a single hyphen in the config doesn't result in potentially hundreds of host groups with incorrect names.

Reply:
These four lines just remove a single extra seperator though? The later part ensures that it's present. But like you say, it doesn't really add much complexity. I still think that less code/less implicit stuff happening is a good thing though.

I agree that the prefix probably should be formalized. I'd go as simple as possible. Maybe add the separator to config. Say that anything before the first occurrence of the separator is the prefix part. You probably want to disallow/raise exceptions if a prefix contains the separator, or you might see confusing stuff happen.

Reply:
I see that `f"{prefix}{sep}{postfix}"` was changed to `f"{prefix}{postfix}"` which of course might will/might remove the separator. I didn't really mean that this part should be changed, just the extra seperator check.

Reply:
Yes, but that needed to change, since otherwise we would _always_ get double dash, seeing as our variables would be `prefix="Foo-"`, `separator="-"`, `suffix="bob-hosts1"`. 

The config specifies that prefixes should be specified _with_ a trailing dash, and as such we will always end up with names such as `Foo-bob-hosts1`, because we are joining the two strings `"Foo-"` and `"bob-hosts1"`.

The existing check that was removed, was just to ensure we always had the separator _somewhere_ between these two strings. I did not change any of the tests, so this new behavior mirrors the old whenever the prefixes contain a trailing dash. 

The new behavior, as a result of this change is that a config containing `extra_siteadmin_hostgroup_prefixes = ["Templates"]` will result in the string `Templatesbob-hosts1`, because that check was removed.

Reply:
I don't think the separator should be allowed in the prefix. I.e. you should just have `"Foo"` in config. This kind of shows that a formal definition regarding how this works probably should be written.